src/cmiss_petsc.f90

```diff
@@ -5601,7 +5635,36 @@ SUBROUTINE PETSC_TSSETFROMOPTIONS(TS_,ERR,ERROR,*)
     CALL EXITS("PETSC_TSSETFROMOPTIONS")
     RETURN 1
   END SUBROUTINE PETSC_TSSETFROMOPTIONS
+
+  !
+  !================================================================================================================================
+  !
     
+  !>Buffer routine to the PETSc TSSetExactFinalTime routine.
+  SUBROUTINE PETSC_TSSETEXACTFINALTIME(TS_,EFTOPT,ERR,ERROR,*)
+
+    TYPE(PETSC_TS_TYPE), INTENT(INOUT) :: TS_ !<The TS to set the initial time step for
+    LOGICAL, INTENT(IN) :: EFTOPT !<The option for exact final time to set
```

Comment:
This shouldn't be logical. It will need to take in the various parameters and call appropriately.